homeassistant/components/sensor/__init__.py

```diff
@@ -587,7 +587,7 @@ def state(self) -> Any:
                     f"state class {state_class} unit {unit_of_measurement} and "
                     f"suggested precision {suggested_precision} thus indicating it "
                     f"has a numeric value; however, it has the non-numeric value: "
-                    f"{value} ({type(value)})"
+                    f"'{value}' ({type(value)})"
```

Comment:
I've often see it blank unit of measurement also (empty string, or space).

Maybe do it for all items? device_class/state_class/unit_of_measurement/suggested_precision/value